src/main/java/org/ice4j/ice/harvest/MappingCandidateHarvesters.java

```diff
@@ -283,42 +283,53 @@ public StunMappingCandidateHarvester call()
 
         // Now run discover() on all created harvesters in parallel and pick
         // the ones which succeeded.
-        ExecutorService es = Executors.newFixedThreadPool(tasks.size());
+        ExecutorService es = Executors.newFixedThreadPool(tasks.size(),
+            runnable -> {
+                Thread thread = new Thread(runnable);
```

Comment:
You may want to set this thread as a daemon thread. See the explanation here: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#defaultThreadFactory()